crates/hyperdrive-math/src/long/targeted.rs

```diff
@@ -0,0 +1,503 @@
+                .await?;
+
+            // Some of the checkpoint passes and variable interest accrues.
+            alice
+                .checkpoint(alice.latest_checkpoint().await?, None)
+                .await?;
+            let variable_rate = rng.gen_range(fixed!(0)..=fixed!(0.5e18));
+            alice
+                .advance_time(
+                    variable_rate,
+                    FixedPoint::from(config.checkpoint_duration) * fixed!(0.5e18),
+                )
+                .await?;
+
+            // Bob opens a targeted long.
+            let max_spot_price_before_long = bob.get_state().await?.calculate_max_spot_price();
+            let target_rate = initial_fixed_rate / fixed!(2e18);
+            let targeted_long = bob
+                .calculate_targeted_long(
+                    target_rate,
+                    Some(num_newton_iters),
+                    Some(allowable_rate_error),
+                )
+                .await?;
+            bob.open_long(targeted_long, None, None).await?;
+
+            // Three things should be true after opening the long:
+            //
+            // 1. The pool's spot price is under the max spot price prior to
+            //    considering fees
+            // 2. The pool's solvency is above zero.
+            // 3. IF Bob's budget is not consumed; then new rate is close to the target rate
+
+            // Check that our resulting price is under the max
+            let spot_price_after_long = bob.get_state().await?.calculate_spot_price();
+            assert!(
+                max_spot_price_before_long > spot_price_after_long,
+                "Resulting price is greater than the max."
+            );
+
+            // Check solvency
+            let is_solvent = {
+                let state = bob.get_state().await?;
+                state.calculate_solvency() > allowable_solvency_error
+            };
+            assert!(is_solvent, "Resulting pool state is not solvent.");
+
+            // If the budget was NOT consumed, then we assume the target was hit.
+            let new_rate = bob.get_state().await?.calculate_spot_rate();
+            if !(bob.base() <= allowable_budget_error) {
+                // Actual price might result in long overshooting the target.
+                let abs_error = if target_rate > new_rate {
+                    target_rate - new_rate
+                } else {
+                    new_rate - target_rate
+                };
+                assert!(
+                    abs_error <= allowable_rate_error,
+                    "target_rate was {}, realized rate is {}. abs_error={} was not <= {}.",
+                    target_rate,
+                    new_rate,
+                    abs_error,
+                    allowable_rate_error
+                );
+
+            // Else, we should have undershot,
+            // or by some coincidence the budget was the perfect amount
+            // and we hit the rate exactly.
+            } else {
```

Comment:
This is just personal preference, so feel free to disregard. To comment `else` blocks, I think this style is pretty nice:

```
// Comment for if
if blah {
    // blah
}
// Comment for else
else {
    // blah
}
```

Reply:
This is what I'm doing, I think. Are you suggesting I remove the newline before the else comment?

Reply:
Oh, I was suggesting that instead of “} else {“ you put the first brace on the line above the comment

Reply:
AHHH ok got it!